parlai/crowdsourcing/tasks/model_chat/analysis/compile_results.py

```diff
@@ -0,0 +1,522 @@
+                        'agent_idx',
+                        'text',
+                    ]
+                    + self.PROBLEM_BUCKETS,
+                )
+                additional_context = ''
+                if data['context_dataset'] == 'wizard_of_wikipedia':
+                    additional_context = '\n' + data['additional_context']
+                df = df.append(
+                    {
+                        'folder': info_dict['read_folder_name'],
+                        'worker_id': info_dict['worker'],
+                        'hit_id': info_dict['hit_id'],
+                        'model_nickname': model_nickname,
+                        'conversation_idx': conversation_idx,
+                        'turn_idx': -1,
+                        'agent_idx': 1,
+                        'text': 'your persona: '
+                        + data['personas'][1][0]
+                        + '\nyour persona: '
+                        + data['personas'][1][1]
+                        + additional_context,
+                        **{bucket: '' for bucket in self.PROBLEM_BUCKETS},
+                    },
+                    ignore_index=True,
+                )
+
+                total_utterances += len(
+                    [d for d in data["dialog"] if d["agent_idx"] == 1]
+                )
+                if len(data['dialog']) > 20:
+                    print(
+                        f'Got long dialogue of {len(data["dialog"])} utterances, hit id: {info_dict["hit_id"]}, model_nickname: {model_nickname}.'
+                    )
+
+                dialog_has_problems = False
+                for utterance_idx, utt in enumerate(data['dialog']):
+                    d = {
+                        'folder': info_dict['read_folder_name'],
+                        'worker_id': info_dict['worker'],
+                        'hit_id': info_dict['hit_id'],
+                        'model_nickname': model_nickname,
+                        'conversation_idx': conversation_idx,
+                        'turn_idx': utterance_idx,
+                        'agent_idx': utt['agent_idx'],
+                        'text': utt['text'],
+                        **{bucket: '' for bucket in self.PROBLEM_BUCKETS},
+                    }
+                    if utt['agent_idx'] == 1:
+                        if 'problem_data' not in utt:
+                            for bucket in self.PROBLEM_BUCKETS:
+                                d[bucket] = 'MALFORMED'
+                            print(
+                                f'Warning got MALFORMED utterance problem data inside complete convo: {utt}. Skipping.'
+                            )
+                            continue
+                        else:
+                            for bucket in self.regular_buckets:
+                                d[bucket] = utt['problem_data'][bucket]
+                            d['none_all_good'] = utt['problem_data']['none_all_good']
```

Comment:
why is this bucket handled separately?

Reply:
Good catch, merged this into the previous line